arcgis-ios-sdk-samples/Augmented reality/Display scenes in tabletop AR/DisplayScenesInTabletopAR.swift

```diff
@@ -0,0 +1,239 @@
+
+    // Wait for at least one detected plane before allowing user to place map
+    var hasFoundPlane = false
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Configure a starting invisible scene with a tiling scheme matching that of the scene that will be used
+        arView.sceneView.scene = AGSScene(tilingScheme: .geographic)
+        arView.sceneView.scene?.baseSurface?.opacity = 0
+
+        // Listen for tracking state changes
+        arView.arSCNViewDelegate = self
+        
+        //add the source code button item to the right of navigation bar
+        (self.navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["DisplayScenesInTabletopAR"]
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        arView.startTracking(.ignore)
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+        arView.stopTracking()
+    }
+
+    private func configureSceneForAR() {
+        // Load the package
+        package.load { [weak self] (err: Error?) in
+            guard let self = self else { return }
+
+            if let error = err {
+                self.presentAlert(error: error)
+            } else if let scene = self.package.scenes.first {
+                // Display the scene
+                self.arView.sceneView.scene = scene
+
+                // Remember that the scene has already been placed
+                self.hasPlacedScene = true
+
+                // Configure scene surface opacity and navigation constraint
+                if let surface = scene.baseSurface {
+                    surface.opacity = 0
+                    surface.navigationConstraint = .none
+                }
+
+                self.updateTranslationFactorAndOriginCamera(scene)
+            }
+        }
+    }
+
+    private func updateTranslationFactorAndOriginCamera(_ scene: AGSScene) {
+        // Create the origin camera to be at the bottom and in the center of the scene
+        // and set the pitch to be 90.0, to match ARKit tracking values
+        let newCam = AGSCamera(latitude: 39.95787000283599,
+                               longitude: -75.16996728256345,
+                               altitude: 8.813445091247559,
+                               heading: 0,
+                               pitch: 90,
+                               roll: 0)
+
+        // Set the origin camera
+        self.arView.originCamera = newCam
+
+        // Scene width is about 800m
+        let geographicContentWidth = 800.0
+
+        // Physical width of the table area the scene will be placed on in meters
+        let tableContainerWidth = 5.0
+
+        // Set the translation factor based on scene content width and desired physical size
+        self.arView.translationFactor = geographicContentWidth / tableContainerWidth
+
+        // Set the origin camera
+        self.arView.originCamera = newCam
```

Comment:
Just noticed that origin camera is set twice. Also, it looks like the scene parameter isn't necessary for this func; I think it was a holdover from an earlier version where the size was calculated programmatically.